source/_posts/2024-02-21-voice-chapter-6.markdown

```diff
@@ -0,0 +1,135 @@
+
+* "okay nabu"
+* "hey jarvis"
+* "alexa"
+
+You can try these out yourself now by following the [ESP32-S3-BOX tutorial][s3-box-tutorial]. Changing the default "okay nabu" wake word will require adjusting your [ESPHome configuration](https://beta.esphome.io/components/micro_wake_word.html) and recompiling the firmware, which may take a long time and requires a machine with more than 2GB of RAM.
+
+We're grateful to Kevin for developing microWakeWord, and making it a part of the open home!
+
+## Sentence trigger responses
+
+Adding custom sentences to Assist is as easy as adding a [sentence trigger][sentence triggers] to an automation. This allows you to trigger any action in Home Assistant with whatever sentences you want.
+
+Now with the new [conversation response] action in HA 2024.2, you can also customize the response spoken or printed back to you. Using [templating](/docs/automation/templating/#sentence), your response can refer to the current state of your home.
+
+<p><img src='/images/blog/2024-02-21-voice-chapter-6/assist-custom-response-editor.png' class='no-shadow' /></p>
+
+You can also refer to [wildcards](/docs/automation/trigger/#sentence-wildcards) in your sentence trigger. For example, the sentence trigger:
+
+```
+play {album} by {artist}
+```
+
+could have the response:
+
+{% raw %}
+```
+Playing {{ trigger.slots.album }} by {{ trigger.slots.artist }}
+```
+{% endraw %}
+
+in addition to calling a media service.
+
+## Improved errors and debugging
+
+Assist users know the phrase "Sorry, I couldn't understand that" all too well. This generic error message was given for a variety of reasons, such as:
+
+* The sentence didn't match any known [intent](https://github.com/home-assistant/intents)
+* The device/area names didn't match
+* There weren't any devices of a specific type in an area (lights, windows, etc.)
+
+Starting in HA 2024.2, Assist provides different error messages for each of these cases.
+
+<p><img src='/images/blog/2024-02-21-voice-chapter-6/error_messages.png' class='no-shadow' /></p>
+
+Now if you encounter errors, you will know where to start looking! The first thing to check is that your device is [exposed to Assist](/voice_control/voice_remote_expose_devices/). Some types of devices, such as lights, are exposed by default. Other, like locks, are not and must be manually exposed.
+
+Once your devices are exposed, make sure you've added an appropriate [alias](/voice_control/aliases) so Assist will know exactly how you'll be referring to them. Devices and areas can have multiple aliases, even in multiple languages, so everyone's preference can be accommodated.
+
+If you are still having problems, the [Assist debug tool][debugging tools] has also been improved. Using the tool, you see how Assist is interpreting a sentence, including any missing pieces.
+
+<p><img src='/images/blog/2024-02-21-voice-chapter-6/debug_tool.png' class='no-shadow' /></p>
```

Comment:
If possible, let's try to always be contextual when dropping screenshots.
A new user may be puzzled by such screenshot and wonder where to find such dev tools.

Here is a replacement screenshot is the full screen (and nice shadows)

![CleanShot 2024-02-21 at 09 19 31](https://github.com/home-assistant/home-assistant.io/assets/5878296/7a2a45c6-6657-4737-b6b5-50a21344a25f)


Let's also add this markdown that will redirect any user to their own Assist devtool:
[![Open your Home Assistant instance and show your Assist developer tools.](https://my.home-assistant.io/badges/developer_assist.svg)](https://my.home-assistant.io/redirect/developer_assist/)
```
[![Open your Home Assistant instance and show your Assist developer tools.](https://my.home-assistant.io/badges/developer_assist.svg)](https://my.home-assistant.io/redirect/developer_assist/)
 ```